### ISSUE TITLE
[feature request] Focus checker

### ISSUE THREAD

**Issue by rpawlaszek** [MayGo/tockler#108]
I've been using this awesome app since December and found out that every now and then, when I have some other issue to attend to, I omit pausing working a specific subject. 

What I would definitely see as a benefit (at least in my case) is a functionality that forces me to respond that I continue working on a specific task. If I could set a sort of 'minimum time window', 'focus check alert' setting to, say, 15mins this would allow me to make sure that my lack of discipline in pausing the monitored task.

Do you think @MayGo that such a functionality is in reach within this application?

**Comment by MayGo** [comment]
This is something that can be added.
Would this alert be displayed after stopping the task + 15 min after? 
Or should it always alert in 15 min interval.
For example: In the morning open up laptop. And after 15 min should notify that I don't have any task running.
If I ignore that notification, should it notify me again after 15 min?

Or should we add "pause task" feature and with that we add 15 min reminder?
What should the message say?
"15 min" will be configurable. And have to explicitly enable to be used.

**Comment by rpawlaszek** [comment]
I had something different in mind, but what you're proposing is great. And I think this could be combined in one.

My suggestion was there because I often start a task and either need to attend to other things (I work remotely now, so it is sometimes as trivial as being called by my kid) or by the end of the day I simply forget to pause the task. And this distorts my estimates of work. That is why I called it 'focus checker' as it would periodically check if I was still working on a task. If I didn't click 'Yes' (or whatever it would look like) then Tockler would automatically pause the task.

But your proposition (if possible) is complementary to this, as it would notify me that I forgot to set a task, so it would be proactive (and when I thought of it - I did also forget to set the task). 

Thus, if this is possible I would rather name it something like... 'Activity assistant'. And if this option is turned on it could:

1. Check if there is any task runnning and if not (idle check) - show a popup with notification
2. Check if the task running is still actually being worked on (focus check) - show a popup with a simple button 'Still working on [App] [Title]' or something like this.

What do you think, @MayGo?

**Comment by MayGo** [comment]
That sounds good. Will try to implement this next month.

**Comment by rpawlaszek** [comment]
Just in clarify: with the second point: if the user does not perform the required action (like clicking "I'm still working on that task") within a specified time (e.g. 30 seconds) the task should be paused and then the application would be notifying that no task is running.

**Comment by rpawlaszek** [comment]
I thought of setting a time that would be a 'minimal grain of resolution' for the tasks. So in my view it should be only one period time set, and I thought of 15 minutes. If within these 15 minutes I do what I do - it'll be included within the 'task activities'.  I wouldn't distinguish that. I think that after some time it will help (here: me) to grow the habit of being aware of the tasks. And even if I need to go away for a minute - this is ok, I would see it as a 'natural way' that I do tasks. I won't be sitting and typing all along. There will be distractions (a mail, somebody messaged me, etc.). Even if I go make myself a tea - fine. So I'd go with a periodic check only and wouldn't bother whether the cause is internal or external. If I lost track on the task then due to the fact that I have the period set - it will show a pop up and I will have lost only maximum 15 minutes in my estimates. As typically (my) tasks are in days - it will not affect the estimates that much.

I think that how you created the app is also very good for personal analysis of how I used the time to actually execute the task (due to having the stats about the applications running). My priority with this app is to help myself estimate how efficient I'm in setting task deadlines (because I do prior to using this app) and then showing how much time I actually spend on a task.

To sum up: my request was for allowing me to grow a habit of keeping the tempo and having a rhythm while working.

**Comment by MayGo** [comment]
Sorry, I haven't had time (motivation) to work on this. I hope this comes in near future, perhaps when I manage to add some paid option to Tockler.

**Comment by rpawlaszek** [comment]
@MayGo, this is understood and thank you for honesty. I do think that this application deserves not only praise, but more importantly - traction. Fingers crossed for transitioning towards a properly paid version, then.

**Comment by herrbrixter** [comment]
It feels like this feature request is scope creep. I like that Tockler is focused on tracking app usage. Sending alerts to check if the user is still working on a task feels like another app. I won't use this feature. My 2 cents.
